src/operator/nn/mkldnn/mkldnn_convolution.cc

```diff
@@ -39,7 +39,7 @@ DMLC_REGISTER_PARAMETER(MKLDNNConvParam);
 bool SupportMKLDNNConv(const ConvolutionParam& params, const NDArray& input) {
   if ((params.kernel.ndim() != 1) && (params.kernel.ndim() != 2) && (params.kernel.ndim() != 3))
```

Comment:
BTW: It could be nice to change it.
```suggestion
  if (params.kernel.ndim() > 3)) return false;
```